scylla/src/transport/connection_pool.rs

```diff
@@ -80,6 +90,43 @@ enum PoolConnections {
     },
 }
 
+fn format_connection_vector(conns: &[Arc<Connection>]) -> String {
+    conns
+        .iter()
+        .map(|conn| conn.get_connect_address().to_string())
+        .collect::<Vec<String>>()
+        .join(",")
+}
+
+fn format_sharded_connection_vector(conns: &[Vec<Arc<Connection>>]) -> String {
+    conns
+        .iter()
+        .enumerate()
+        .map(|(shard_no, conn_vec)| format!("{}: {}", shard_no, format_connection_vector(conn_vec)))
+        .collect::<Vec<String>>()
+        .join("; ")
+}
+
+impl std::fmt::Display for PoolConnections {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            PoolConnections::NotSharded(conns) => {
+                write!(f, "non-sharded: {}", format_connection_vector(conns))
+            }
+            PoolConnections::Sharded {
+                sharder,
+                connections,
+            } => write!(
+                f,
+                "sharded({};{}): {}",
```

Comment:
The message format looks a bit cryptic to the outsider (aka. user). How about this:

```
sharded(nr_shards: {}, msb_ignore_bits: {}): ...
```

The `...` part, for which I was too lazy to come up with a precise format string, we could include information about how many connections are there for each shard.